eclass/latex-package.eclass

```diff
@@ -105,79 +94,73 @@ latex-package_src_doinstall() {
 
 	# This actually follows the directions for a "single-user" system
 	# at http://www.ctan.org/installationadvice/ modified for gentoo.
-	[ -z "$1" ] && latex-package_src_install all
+	[[ -z ${1} ]] && latex-package_src_install all
 
-	while [ "$1" ]; do
-		case $1 in
+	while [[ ${1} ]]; do
+		case ${1} in
 			"sh")
-				for i in `find . -maxdepth 1 -type f -name "*.${1}"`
+				find . -maxdepth 1 -type f -name "*.${1}" | while read i;
 				do
-					dobin $i || die "dobin $i failed"
+					dobin ${i}
 				done
 				;;
 			"sty" | "cls" | "fd" | "clo" | "def" | "cfg")
-				for i in `find . -maxdepth 1 -type f -name "*.${1}"`
+				find . -maxdepth 1 -type f -name "*.${1}" | while read i;
 				do
 					insinto ${TEXMF}/tex/latex/${PN}
-					doins $i || die "doins $i failed"
+					doins ${i}
 				done
 				;;
 			"dvi" | "ps" | "pdf")
-				for i in `find . -maxdepth 1 -type f -name "*.${1}"`
+				find . -maxdepth 1 -type f -name "*.${1}" | while read i;
 				do
 					insinto /usr/share/doc/${PF}
-					doins $i || die "doins $i failed"
+					doins ${i}
 					dosym /usr/share/doc/${PF}/$(basename ${i}) ${TEXMF}/doc/latex/${PN}/${i}
-					case "${EAPI:-0}" in
-						0|1|2|3) ;;
-						*)
-							# prevent compression of symlink target
-							docompress -x /usr/share/doc/${PF}/$(basename ${i})
-							;;
-					esac
+					docompress -x /usr/share/doc/${PF}/$(basename ${i})
 				done
 				;;
 			"tex" | "dtx")
 				if ! in_iuse doc || use doc ; then
-					for i in `find . -maxdepth 1 -type f -name "*.${1}"`
+					find . -maxdepth 1 -type f -name "*.${1}" | while read i;
 					do
-						[ -n "${LATEX_PACKAGE_SKIP}" ] && has ${i##*/} ${LATEX_PACKAGE_SKIP} && continue
-						einfo "Making documentation: $i"
-						if pdflatex ${LATEX_DOC_ARGUMENTS} --halt-on-error --interaction=nonstopmode $i ; then
-							pdflatex ${LATEX_DOC_ARGUMENTS} --halt-on-error --interaction=nonstopmode $i || die
+						[[ -n ${LATEX_PACKAGE_SKIP} ]] && has ${i##*/} ${LATEX_PACKAGE_SKIP} && continue
+						einfo "Making documentation: ${i}"
+						if pdflatex ${LATEX_DOC_ARGUMENTS} --halt-on-error --interaction=nonstopmode ${i} ; then
+							pdflatex ${LATEX_DOC_ARGUMENTS} --halt-on-error --interaction=nonstopmode ${i} || die
```

Comment:
Are you deliberately calling the same command twice? If so, leave an explicit comment because right now, it looks like the first call was supposed to be some kind of 'pretend' while it doesn't have any argument for that.

Reply:
I guess it s an exception made on purpose, sometimes pdflatex must be called twice.